src/CMakeLists.txt

```diff
@@ -27,7 +27,15 @@ if (WIN32)
 endif()
 
 if(INPUTLEAP_BUILD_TESTS)
-    include(../cmake/gtest.cmake)
+    include(FetchContent)
+    set(INSTALL_GTEST OFF)
+    FetchContent_Declare(
```

Comment:
I think fetching code via cmake instead of using a submodule would be a regression in usability.

Reply:
It only fetches if it's not installed

Reply:
Why do you think this would be a regression in usability?

Reply:
Currently the repository carries a specific version of gtest together and there's no need to use any additional tools to get full source code. One can do `git submodule update --init` and then browse the code without running cmake. After this change the code will be managed by cmake which is I think not the best tool for this.

I do understand that in some cases cmake may be better and some people will find it more convenient just because they are more used to it. Other people will say the same about git submodules. In the end a choice needs to be made and as a maintainer I choose git submodules.

Reply:
I don't know, why would you browse the code of the submodule?
Seems much less elegant this way
But at least you can merge the commit for gulrak, put it to 
https://github.com/input-leap/input-leap/pull/2010
The other commits we can still discuss

Reply:
> I don't know, why would you browse the code of the submodule?

I do need to do that sometimes. Anyway, let's put all technical reasons aside and say this is a matter of personal preference. Being a maintainer I get to choose what's more convenient to myself :-)

I would prefer that our time is spent more on improving InputLeap and not arguing about things that don't matter in the end. Both submodules and cmake work, maintainer does not like the latter, means we just leave the matter as is.

Reply:
I deliberately put technical reasons aside because if we look to large projects, they tend to use weird setups due to "reasons". Webkit copy-pastes all third-party code to within git repo, whereas Chromium has 250 first-level git submodules.

Reply:
> I think fetching code via cmake instead of using a submodule would be a regression in usability.

Agreed. Who in their right mind thinks this is a better solution than using submodules?

Submodules are the de-facto way to include dependencies that need to be built from source or included as a library that aren't commonly available in binary from on the intended targets.

Reply:
Though in this case google test are commonly available in binary form, and building them from source is more of a fallback. In that case my (own) personal preference would be to use some source distribution as allowed by cmake rather than embedding the whole development history of google test – there’s no real co-versioning going on. But I also don’t think it really matters.

Reply:
> rather than embedding the whole development history of google test

that is what `shallow = true` exists for `.gitmodules`. Just add it and problem solved.

Reply:
https://github.com/input-leap/input-leap/blob/bca5258e5edfa8105d46b6ca42ea550b3e80ef10/.gitmodules#L1-L3

Reply:
>  Just add it
^

Reply:
> that is what shallow = true exists for .gitmodules. Just add it and you won't clone any history by default but you will still link to a particular commit/branch/tag.

Cool, I didn't know about it. This is great idea.